src/cpp/src/whisper_pipeline.cpp

```diff
@@ -41,21 +45,17 @@ std::vector<int64_t> whisper_generate(const ov::genai::WhisperGenerationConfig&
                                       ov::genai::WhisperFeatureExtractor& feature_extractor,
                                       const std::shared_ptr<StreamerBase> streamer);
 
-class WhisperPipeline::Impl {
+class WhisperPipelineStatefulImpl : public WhisperPipelineImplBase {
```

Comment:
@ilya-lavrenov is `WhisperPipelineStatefulImpl` correct naming? Models aren't stateful, they expose KV-cache outside

Reply:
good point.

Let's keep name as is and rename to stateful as they really become stateful